metrics/prometheus/prometheus.go

```diff
@@ -1,196 +1,202 @@
+		dtoQuantiles := make([]*dto.Quantile, len(quantiles))
+		for i := range thresholds {
+			dtoQuantiles[i] = &dto.Quantile{
+				Quantile: ptrTo(quantiles[i]),
+				Value:    ptrTo(thresholds[i]),
 			}
+		}
 
-			ps := snapshot.Percentiles(pvShortPercent)
-			qs := make([]*dto.Quantile, len(pv))
-			for i := range pvShort {
-				v := pv[i]
-				s := ps[i]
-				qs[i] = &dto.Quantile{
-					Quantile: &v,
-					Value:    &s,
-				}
+		return &dto.MetricFamily{
+			Name: &name,
+			Type: dto.MetricType_SUMMARY.Enum(),
+			Metric: []*dto.Metric{{
+				Summary: &dto.Summary{
+					SampleCount: ptrTo(uint64(snapshot.Count())), //nolint:gosec
+					SampleSum:   ptrTo(float64(snapshot.Sum())),
+					Quantile:    dtoQuantiles,
+				},
+			}},
+		}, nil
+	case metrics.Meter:
+		return &dto.MetricFamily{
+			Name: &name,
+			Type: dto.MetricType_GAUGE.Enum(),
+			Metric: []*dto.Metric{{
+				Gauge: &dto.Gauge{
+					Value: ptrTo(float64(m.Snapshot().Count())),
+				},
+			}},
+		}, nil
+	case metrics.Timer:
+		snapshot := m.Snapshot()
+
+		quantiles := []float64{.5, .75, .95, .99, .999, .9999}
+		thresholds := snapshot.Percentiles(quantiles)
+		dtoQuantiles := make([]*dto.Quantile, len(quantiles))
+		for i := range thresholds {
+			dtoQuantiles[i] = &dto.Quantile{
+				Quantile: ptrTo(quantiles[i]),
+				Value:    ptrTo(thresholds[i]),
 			}
+		}
 
-			mfs = append(mfs, &dto.MetricFamily{
-				Name: &name,
-				Type: dto.MetricType_SUMMARY.Enum(),
-				Metric: []*dto.Metric{{
-					Summary: &dto.Summary{
-						SampleCount: &count,
-						// TODO: do we need to specify SampleSum here? and if so
-						// what should that be?
-						Quantile: qs,
-					},
-				}},
-			})
+		return &dto.MetricFamily{
+			Name: &name,
+			Type: dto.MetricType_SUMMARY.Enum(),
+			Metric: []*dto.Metric{{
+				Summary: &dto.Summary{
+					SampleCount: ptrTo(uint64(snapshot.Count())), //nolint:gosec
+					SampleSum:   ptrTo(float64(snapshot.Sum())),
+					Quantile:    dtoQuantiles,
+				},
+			}},
+		}, nil
+	case metrics.ResettingTimer:
+		snapshot := m.Snapshot()
+		if snapshot.Count() == 0 {
+			return nil, fmt.Errorf("%w: %q resetting timer metric count is zero", errMetricSkip, name)
 		}
-	}
 
-	return mfs, nil
-}
+		pvShortPercent := []float64{50, 95, 99}
+		thresholds := snapshot.Percentiles(pvShortPercent)
+		dtoQuantiles := make([]*dto.Quantile, len(pvShortPercent))
+		for i := range pvShortPercent {
+			dtoQuantiles[i] = &dto.Quantile{
+				Quantile: ptrTo(pvShortPercent[i]),
+				Value:    ptrTo(thresholds[i]),
+			}
+		}
 
-func Gatherer(reg metrics.Registry) prometheus.Gatherer {
-	return gatherer{reg: reg}
+		return &dto.MetricFamily{
+			Name: &name,
+			Type: dto.MetricType_SUMMARY.Enum(),
+			Metric: []*dto.Metric{{
+				Summary: &dto.Summary{
+					SampleCount: ptrTo(uint64(snapshot.Count())), //nolint:gosec
+					Quantile:    dtoQuantiles,
+				},
+			}},
+		}, nil
+	default:
+		return nil, fmt.Errorf("metric %q: type is not supported: %T", name, metric)
```

Comment:
this won't cause any unknown geth metric type to break this, right? (e.g a new upstream type in the future)

Reply:
Shouldn't it break it? This would show in the CI right, and we should add the type to our switch case and either handle it or explicitly ignore it, I think.

Reply:
The only place that gatherer is registered is:

https://github.com/ava-labs/subnet-evm/blob/4f2ea6381cb6ae0dd911cf46f5d295b3291ee7a4/plugin/evm/vm.go#L558

Which registers it to [avalanchego's multi gatherer](https://github.com/ava-labs/avalanchego/blob/9c9b0042c76f2075f99faa478489bab9fed7e011/snow/context.go#L54) and all errors are checked/none-are-discarded.

Maybe it's not covered by tests, but it should definitely show up in e2e tests (and, if not, we would have an error handling problem in avalanchego).